language-support/java/codegen/BUILD.bazel

```diff
@@ -123,6 +125,16 @@ daml_compile(
     target = "1.13",
 )
 
+daml_compile(
+    name = "test-template",
+    srcs = ["src/test/daml/Bar.daml"],
+)
+
+daml_compile(
+    name = "test-template2",
+    srcs = ["src/test/daml/Bar.daml"],
+)
```

Comment:
There's an important integration test to add, too:

1. One dar with prefix `pkg1` and a template that uses for its field a `data` type defined in another dar with prefix `pkg2`
2. The first dar should incorporate the second by means of `data_dependencies`, not by compiling the sources again; there's an example in the typescript codegen tests
3. Codegen it with the aforementioned prefixes and javac the result
4. Write scala that (statically by means of type ascription) asserts that the field in the template (prefixed by `pkg1`) has the type prefixed by `pkg2`

I have a suspicion that this will not work and you will have to change the design, because the tests in this PR only consider the case where the types under different prefixes can be treated as wholly separate universes.

Reply:
> The first dar should incorporate the second by means of data_dependencies, not by compiling the sources again; there's an example in the typescript codegen tests

Can you point me to the example of typescript codegen tests, please?

Reply:
It is intended, to get wholly separate universes when using prefixes.

Reply:
Is there a specific use case that we are misunderstanding?

Reply:
> Can you point me to the example of typescript codegen tests, please?

@chunlokling-da

https://github.com/digital-asset/daml/blob/7cac969756a561f54652f35cb9ea540a4f3ea40a/language-support/ts/codegen/tests/BUILD.bazel#L22-L27

> It is intended, to get wholly separate universes when using prefixes...code in pkg1 would not refer to code generated in pkg2, since the prefix was provided for that.

@ray-roestenburg-da How can a prefix setting of dar1=pkg1, dar2=pkg2 to the Java codegen affect whether Daml code in dar1 statically refers to Daml code in dar2?

> Is there maybe a specific use case that we are misunderstanding?

What I've described is a "code in dar1 uses code in dar2" case, which works in `main` when you have disparate prefix settings but I believe will regress with this change.